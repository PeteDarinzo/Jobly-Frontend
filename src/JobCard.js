import React from "react";
import "./Card.css";

const JobCard = ({ title, salary, equity, companyName }) => {

  return (
    <div className="CompanyCard shadow p-3">
      <h2>{title}</h2>
      <p>{companyName}</p>
      <p>Salary: {salary}</p>
      <p>Equity: {equity}</p>
    </div>
  );

}

export default JobCard;