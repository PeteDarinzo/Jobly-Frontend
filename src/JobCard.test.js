import JobCard from "./JobCard";
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom"

it("renders a job card without crashing", () => {
  render(
    <MemoryRouter>
      <JobCard id={1} title="test-job" salary={1000} equity={0} companyName="test-company" apply={() => { }} applied={false} />
    </MemoryRouter>
  );
});

it("matches the home snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <JobCard id={1} title="test-job" salary={1000} equity={0} companyName="test-company" apply={() => { }} applied={false} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});