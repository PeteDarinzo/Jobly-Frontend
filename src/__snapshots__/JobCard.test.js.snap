// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the home snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card shadow p-3"
  >
    <h2>
      test-job
    </h2>
    <p>
      test-company
    </p>
    <p>
      Salary: 1000
    </p>
    0
    <button
      class="btn-lg m-2 btn btn-danger"
      type="button"
    >
      Apply
    </button>
  </div>
</DocumentFragment>
`;
